<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GrowthComputationDlg</class>
 <widget class="QDialog" name="GrowthComputationDlg">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>778</width>
    <height>571</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Images selection</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Available images</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Selected images</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QTreeView" name="allImages">
        <property name="contextMenuPolicy">
         <enum>Qt::CustomContextMenu</enum>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::ExtendedSelection</enum>
        </property>
        <property name="iconSize">
         <size>
          <width>64</width>
          <height>64</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="addImages">
          <property name="minimumSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="icons.qrc">
            <normaloff>:/icons/right.png</normaloff>:/icons/right.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="removeImages">
          <property name="minimumSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="icons.qrc">
            <normaloff>:/icons/left.png</normaloff>:/icons/left.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item row="1" column="2">
       <widget class="QTreeView" name="selectedImages">
        <property name="contextMenuPolicy">
         <enum>Qt::CustomContextMenu</enum>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::ExtendedSelection</enum>
        </property>
        <property name="iconSize">
         <size>
          <width>64</width>
          <height>64</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <widget class="QGroupBox" name="groupBox_2">
       <property name="toolTip">
        <string>Criteria under which a cell growth will be computed or not.</string>
       </property>
       <property name="title">
        <string>Cell selection</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QRadioButton" name="samePoints">
            <property name="text">
             <string>Full cells only</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QCheckBox" name="daughterCells">
            <property name="text">
             <string>Include daughter cells</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QRadioButton" name="allCells">
            <property name="text">
             <string>All cells</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QFrame" name="frame">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="frameShape">
             <enum>QFrame::NoFrame</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Plain</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="margin">
              <number>0</number>
             </property>
             <item>
              <widget class="QCheckBox" name="withVariation">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>with</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="maxVariationAllCells">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="suffix">
                <string>%</string>
               </property>
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>200</number>
               </property>
               <property name="value">
                <number>30</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>area variation</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>17</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QRadioButton" name="addPoints">
            <property name="text">
             <string>Add points only</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>with</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="maxVariationAddPoints">
            <property name="suffix">
             <string>%</string>
            </property>
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>200</number>
            </property>
            <property name="value">
             <number>30</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>area variation</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>17</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Method</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QRadioButton" name="forwardMethod">
          <property name="toolTip">
           <string>Compute growth, placing the computation on the start image.</string>
          </property>
          <property name="statusTip">
           <string/>
          </property>
          <property name="text">
           <string>Forward</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QRadioButton" name="backwardMethod">
          <property name="toolTip">
           <string>Compute growth, placing the computation on the end image.</string>
          </property>
          <property name="statusTip">
           <string/>
          </property>
          <property name="text">
           <string>Backward</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>68</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="2">
         <widget class="QSpinBox" name="samplingPoints">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="suffix">
           <string> points</string>
          </property>
          <property name="minimum">
           <number>10</number>
          </property>
          <property name="maximum">
           <number>5000</number>
          </property>
          <property name="singleStep">
           <number>10</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QCheckBox" name="resample">
          <property name="toolTip">
           <string>If checked, the contour of the cells are resampled, associating common vertices but trying to spread point all over on the start cell. Otherwise, only the corresponding vertices are used to compute growht.</string>
          </property>
          <property name="text">
           <string>Resample</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="1" column="0" colspan="2">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>File to save:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="savePath"/>
       </item>
       <item>
        <widget class="QPushButton" name="chooseFile">
         <property name="maximumSize">
          <size>
           <width>32</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close|QDialogButtonBox::Save</set>
     </property>
    </widget>
   </item>
  </layout>
  <action name="actionSelect_even_images">
   <property name="text">
    <string>Select even images</string>
   </property>
  </action>
  <action name="actionSelect_odd_images">
   <property name="text">
    <string>Select odd images</string>
   </property>
  </action>
  <action name="actionSelect_all_images">
   <property name="text">
    <string>Select all images</string>
   </property>
  </action>
  <action name="actionSelect_no_image">
   <property name="text">
    <string>Select no image</string>
   </property>
  </action>
  <action name="actionInvert_selection">
   <property name="text">
    <string>Invert selection</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>GrowthComputationDlg</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>565</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>GrowthComputationDlg</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>565</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addPoints</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_4</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>426</y>
    </hint>
    <hint type="destinationlabel">
     <x>187</x>
     <y>428</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addPoints</sender>
   <signal>toggled(bool)</signal>
   <receiver>maxVariationAddPoints</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>426</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>428</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addPoints</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_5</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>73</x>
     <y>426</y>
    </hint>
    <hint type="destinationlabel">
     <x>354</x>
     <y>428</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>withVariation</sender>
   <signal>toggled(bool)</signal>
   <receiver>maxVariationAllCells</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>396</y>
    </hint>
    <hint type="destinationlabel">
     <x>175</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>withVariation</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_2</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>399</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>allCells</sender>
   <signal>toggled(bool)</signal>
   <receiver>withVariation</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>allCells</sender>
   <signal>toggled(bool)</signal>
   <receiver>frame</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>62</x>
     <y>385</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>378</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resample</sender>
   <signal>toggled(bool)</signal>
   <receiver>samplingPoints</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>565</x>
     <y>372</y>
    </hint>
    <hint type="destinationlabel">
     <x>666</x>
     <y>372</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
